app/src/main/java/com/gatheringhallstudios/mhworlddatabase/adapters/LocationAdapterDelegate.kt

```diff
@@ -0,0 +1,36 @@
+package com.gatheringhallstudios.mhworlddatabase.adapters
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+import com.gatheringhallstudios.mhworlddatabase.R
+import com.gatheringhallstudios.mhworlddatabase.common.SimpleListDelegate
+import com.gatheringhallstudios.mhworlddatabase.data.views.LocationView
+import com.gatheringhallstudios.mhworlddatabase.getAssetDrawable
+import kotlinx.android.synthetic.main.listitem_location.view.*
+
+import kotlinx.android.synthetic.main.listitem_monster.view.*
+
+class LocationAdapterDelegate(private val onSelected: (LocationView) -> Unit)
+    : SimpleListDelegate<LocationView, View>() {
+
+    override fun getDataClass() = LocationView::class
+
+    override fun onCreateView(parent: ViewGroup): View {
+        val inflater = LayoutInflater.from(parent.context)
+        return inflater.inflate(R.layout.listitem_monster, parent, false)
```

Comment:
For normal ListItems we usually try to use the generic `IconLabelTextCell` component. Like this
```
            val view = IconLabelTextCell(context)
            val icon = ContextCompat.getDrawable(context!!, R.drawable.question_mark_grey)

            view.setLeftIconDrawable(icon)
            view.setLabelText(habitat.location_name)
            view.setValueText(areas.toString())
``` 

For Locations I realized that we don't have a similar, generic component for slightly larger ListItems like Monsters and Locations. Can you please turn the `listitem_monster` layout to a generic listitem_large?

Reply:
unfortunately we weren't using it for adapters already even though we should (mybad). But yeah using that object instead is something we really should be starting to do.

Reply:
Deleted the previous since I misunderstood. Jayson is suggesting to turn listitem_monster into listitem_large, and convert both monsters and locations to use it. It would help improve the overall quality of the app.